### PR TITLE
Add possibility to start and stop particle emitter

### PR DESCRIPTION
**Pull request by AlexandreSi**
To bring a solution to this
https://forum.gdevelop-app.com/t/stop-particles/24905

### CODE REVIEW

**Comment by Bouh** [comment]
Think to start a new branch from master, there are expressions IsFlippedX and IsFlippedY on this draft PR.

**Comment by AlexandreSi** [comment]
That's very true, I'll fix that, thanks!

File: Extensions/ParticleSystem/ExtensionSubDeclaration1.cpp [hunk 1]
```diff
@@ -267,4 +267,22 @@ void ExtensionSubDeclaration1(gd::ObjectMetadata& obj) {
                    "CppPlatform/Extensions/particleSystemicon16.png")
       .AddParameter("object", _("Object"), "ParticleEmitter")
       .UseStandardRelationalOperatorParameters("number");
+
+  obj.AddAction("StartEmission",
+                _("Start emission"),
+                _("Refull tank (if not infinite) and start emission of the particles."),
```
Comment (4ian) [inline Extensions/ParticleSystem/ExtensionSubDeclaration1.cpp:273]:
```suggestion
                _("Refill tank (if not infinite) and start emission of the particles."),
```

**Review by 4ian — APPROVED** [review]
🚀

File: Extensions/ParticleSystem/ExtensionSubDeclaration1.cpp (hunk 1 above)
Comment (AlexandreSi) [reply Extensions/ParticleSystem/ExtensionSubDeclaration1.cpp]: Oups, I forgot that I could accept this suggestion directly. Thanks!

**Comment by AlexandreSi** [comment]
@4ian I added 2 commits:
- One that should bring clarity (why restart emitter when changing capacity?) but may need fixes for games that expect the emitter to start when changing its capacity.
- One for code tidiness
Let me know!

**Comment by AlexandreSi** [comment]
But since the Change Capacity action was not working before my previous PR, the first bullet point is ok

**Comment by 4ian** [comment]
> One that should bring clarity (why restart emitter when changing capacity?) 

Isn't this actually a behavior change introduced by https://github.com/4ian/GDevelop/commit/2febbf439f24eb1f5153264c926a044e07109e7c? 
In which case your change here is actually "fixing" this behavior change (not resetting the emission when changing the capacity/flow, which is better).

> One for code tidiness

Maybe we should take the habit of adding the new name next to Deprecated (`SetHidden() // Deprecated - use "SetXxx" instead.`). Otherwise LGTM!

**Comment by 4ian** [comment]
> One that should bring clarity (why restart emitter when changing capacity?) 

Isn't this actually a behavior change introduced by https://github.com/4ian/GDevelop/commit/2febbf439f24eb1f5153264c926a044e07109e7c? 
In which case your change here is actually "fixing" this behavior change (not resetting the emission when changing the capacity/flow, which is better).

> One for code tidiness

Maybe we should take the habit of adding the new name next to Deprecated (`SetHidden() // Deprecated - use "SetXxx" instead.`). Otherwise LGTM!

**Comment by 4ian** [comment]
> One that should bring clarity (why restart emitter when changing capacity?) 

Isn't this actually a behavior change introduced by https://github.com/4ian/GDevelop/commit/2febbf439f24eb1f5153264c926a044e07109e7c? 
In which case your change here is actually "fixing" this behavior change (not resetting the emission when changing the capacity/flow, which is better).

> One for code tidiness

Maybe we should take the habit of adding the new name next to Deprecated (`SetHidden() // Deprecated - use "SetXxx" instead.`). Otherwise LGTM!

**Review by 4ian — APPROVED** [review]

**Pull request merged**